src/func/imports/create-import.ts

```diff
@@ -66,77 +83,130 @@ function createImportDataRecord(transacting: Transaction, dataSets, importData:
+		const languageSet = await updateLanguageSet(orm, transacting, null, languages);
+		dataSets.languageSetId = languageSet && languageSet.get('id');
 	}
 
 	if (releaseEvents) {
-		const releaseEventSet =
-			await updateReleaseEventSet(orm, transacting, null, releaseEvents);
-		entityDataSet.releaseEventSetId =
-			releaseEventSet && releaseEventSet.get('id');
+		const releaseEventSet = await updateReleaseEventSet(orm, transacting, null, releaseEvents);
+		dataSets.releaseEventSetId = releaseEventSet && releaseEventSet.get('id');
 	}
 	// Skipping publisher field, as they're not required in imports.
 
-	return entityDataSet;
+	return dataSets;
 }
 
-export function createImport(orm: ORM, importData: QueuedEntity) {
-	return orm.bookshelf.transaction(async (transacting) => {
+export type ExistingImportAction =
+	| 'skip'
+	| 'update pending'
+	| 'update pending and accepted';
+
+export type ImportOptions = Partial<{
+
+	/** Specifies what should happen if an import with the same external identifier already exists. */
+	existingImportAction: ExistingImportAction;
+}>;
+
+export type ImportStatus =
+	| 'created pending'
+	| 'skipped pending'
+	| 'unchanged pending'
+	| 'updated pending'
+	| 'skipped accepted'
+	| 'unchanged accepted'
+	| 'updated accepted';
+
+export type ImportResult = {
+
+	/** ID of the imported entity (numeric for now, will be a BBID in a future version). */
+	importId: number | string;
+
+	/** Import status of the processed entity. */
+	status: ImportStatus;
+};
+
+export function createImport(orm: ORM, importData: QueuedEntity, {
+	existingImportAction = 'skip'
+}: ImportOptions = {}): Promise<ImportResult> {
+	if (!ENTITY_TYPES.includes(importData.entityType)) {
+		throw new Error('Invalid entity type');
+	}
+
+	return orm.bookshelf.transaction<ImportResult>(async (transacting) => {
 		const {entityType} = importData;
 		const {alias, identifiers, disambiguation, source} = importData.data;
 
-		const [aliasSet, identifierSet, disambiguationObj, entityDataSets] =
+		// Get origin_source
+		let originSourceId: number = null;
+
+		try {
+			originSourceId = await getOriginSourceId(transacting, source);
+		}
+		catch (err) {
+			// TODO: useless, we are only catching our self-thrown errors here
+			throw new Error(`Error during getting source id - ${err}`);
+		}
+
+		const [existingImport] = await getImportMetadata(transacting, originSourceId, importData.originId);
+		if (existingImport) {
+			const isPendingImport = !existingImport.entity_id;
+			if (existingImportAction === 'skip') {
+				return {
+					importId: existingImport.import_id,
+					status: isPendingImport ? 'skipped pending' : 'skipped accepted'
+				};
+			}
+			else if (isPendingImport) {
+				// TODO: update/reuse already existing data of pending imports
+			}
+			else {
+				// TODO: create updates for already accepted entities in a later version
+				return {
+					importId: existingImport.import_id,
+					status: 'skipped accepted'
```

Comment:
Should this be 'updated accepted' ?

Reply:
Yeah it will be, as soon as I've addressed the TODO comment and actually implemented updating for accepted entities. Until then I'm telling the user the truth that I've simply skipped the entity (because accepted entities are not supported yet).
But you're right, this looks a bit confusing... maybe it makes more sense if I temporarily comment out `'update pending and accepted'` from the `ExistingImportAction` type?

Reply:
In addition to the above: Actually I've just realized that I forgot to add a condition `if (existingImportAction === 'update pending')` here to separately handle the case where we only want to create updates for pending entities, but not for accepted entities :see_no_evil:
Will address that tomorrow...

Reply:
d1e048d76eb4e22da85456eefc8bc2a9abf23725 should make it clearer